sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnReflector.java

```diff
@@ -737,14 +430,13 @@ private SimpleDoFnAdapter(DoFnReflector reflector, DoFn<InputT, OutputT> fn) {
 
     @Override
     public void startBundle(OldDoFn<InputT, OutputT>.Context c) throws Exception {
-      ContextAdapter<InputT, OutputT> adapter = new ContextAdapter<>(fn, c);
-      invoker.invokeStartBundle(adapter, adapter);
+      this.fn.prepareForProcessing();
```

Comment:
Calling `prepareForProcessing` should not move here. This class will be going away once OldDoFn disappears. We should not rely on it for invoking methods.